### ISSUE TITLE
Improve accessibility of MooseDoc-generate websites

### ISSUE THREAD

**Issue by cmacmackin** [idaholab/moose#29857]
### Motivation

There exist a number of of [Web Content Accessibility Guidelines](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG) (WCAG) developed by the W3C's [Web Accessibility Initiative](https://www.w3.org/WAI/). These make websites more accessible to those with disabilities by ensuring content has good contrast against the background, pages are navigable using a keyboard, they are well-suited for the use of screen-readers, etc. Browsers such as Firefox can be used to check whether pages meet these standards. This reveals that there are a number of instances where the MOOSE documentation does not meet these guidelines.

### Design

According to Firefox's accessibility checker, the following changes should be made to meet the WCAG.

- ~~[Add alt-text to the images linking to the GitHub repo at the top of the page.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Text_labels_and_names?utm_source=devtools&utm_medium=a11y-panel-checks-text-label#content_with_images_must_be_labeled)~~
- ~~[Remove the `tabindex` attribute from pre-formatted code blocks](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#focusable_elements_should_have_interactive_semantics)~~
- ~~[Increase contrast of navbar breadcrumbs in dark mode](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Color_contrast?utm_source=devtools&utm_medium=a11y-panel-checks-color-contrast)~~
- ~~Increase contrast of table of contents sidebar in dark mode~~
- [Side navigation menu (for narrow screens) should have a link or button to close it, rather than just clicking anywhere.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard?utm_source=devtools&utm_medium=a11y-panel-checks-keyboard#clickable_elements_must_be_focusable_and_should_have_interactive_semantics) (not certain whether this is important)
- ~~Images inserted with the media extension have no alt-text (not clear to me whether this is an issue with the media extension or people just haven't written any in the doc files).~~ Also, these images are clickable even though img and picture tags don't have interactive semantics. It would be better to wrap them in a link or something which does have interactive semantics and attach the zoom feature to that.
- ~~The arrows used to expand/collapse parameter lists in source documentation pages have insufficient contrast in light-mode.~~
- The expandable lists used for parameters are clickable even though lists are not considered interactive elements.
- ~~The headline text for parameters have insufficient contrast in dark-mode.~~

### Impact

These changes will make the MOOSE website easier to read for those with visual impairments and easier to navigate using a keyboard.

**Comment by 00aw** [comment]
I work in higher education and we are required by law to use resources that conform to WCAG 2.1 A/AA success criteria. It would be great if MOOSE was in compliance and we didn't have to restrict use!

**Comment by cmacmackin** [comment]
I've started working on this. I've been looking at the problem with clickable semantics for images. This behaviour uses the [materialbox](https://materializecss.com/media.html) class, which adds a `click` even to the `picture` element. Strictly speaking you shouldn't make a picture clickable (that's not what the element is meant for) but it seems materialbox doesn't support other types of elements with this tag very well. I can get around this by giving the `picture` element the role "button", although this is generally not recommended (it's preferred to use HTML element types with the desired semantics by default, rather than try to force different semantics onto them).

The bigger problem is the action to click away from the zoomed in image. Materialize sets this up using elements without interactive semantics. They are created by the Materialize javascript so there is no way for me to modify them without also modifying Materialize. As such I think we're just going to have to live with that one.

**Comment by cmacmackin** [comment]
A couple more issues I've come across

- ~~Slideshows apparently don't have a <title> tag.~~
- ~~Firefox complained that the arrows in slide shows may not indicate when they are focussed.~~

**Comment by cmacmackin** [comment]
>Remove the tabindex attribute from pre-formatted code blocks

This is referenced in PrismJS/prism#3658 and the reason they chose to do this is explained in PrismJS/prism#2799. See, in particular [this comment](https://github.com/PrismJS/prism/issues/2695#issuecomment-1021734595). I won't change this behaviour.

**Comment by cmacmackin** [comment]
The issues which were practical to fix have now been addressed in #30946 and #30994. Remaining problems have been put into a separate issues #30993.
